src/Authoring/WinRT.SourceGenerator/DiagnosticUtils.cs

```diff
@@ -109,22 +110,20 @@ string QualifiedName(INamedTypeSymbol sym)
                 return sym.OriginalDefinition.ContainingNamespace + "." + sym.OriginalDefinition.MetadataName; 
             }
 
-            bool SameProp(string a, string b)
-            {
-                if (a.StartsWith("get_") || a.StartsWith("put_"))
-                {
-                    return a.Substring(4) == b;
-                }
-                return false;
-            }
-
             foreach (var @interface in typeSymbol.AllInterfaces.
-                        Where(symbol => MappedCSharpTypes.Contains(QualifiedName(symbol)) ||
-                                        ImplementedInterfacesWithoutMapping.Contains(QualifiedName(symbol))))
+                        Where(symbol => WinRTTypeWriter.MappedCSharpTypes.ContainsKey(QualifiedName(symbol)) ||
+                                        WinRTTypeWriter.ImplementedInterfacesWithoutMapping.Contains(QualifiedName(symbol))))
             {
-                var mems = @interface.GetMembers();
-                methods = methods.Where(m => !mems.Where(mem => mem.Name == m.Identifier.Text).Any());
-                properties = properties.Where(p => !mems.Where(mem => SameProp(mem.Name, p.Identifier.Text)).Any()); 
+                var InterfaceMembers = @interface.GetMembers();
+
+                bool ValidMember(MemberDeclarationSyntax member)

```

Comment:
might be performant to do a loop over interface members and build a list of class members using FindImplementationForInterfaceMember and then do a remove all or where not contains.